### PR TITLE
[ADP-3378] Bump node to 8.12

### PR DESCRIPTION
**Pull request by Anviking**
- [x] Bump runtime- and compile-time dependencies to 8.12

### Issue Number

ADP-3381

### CODE REVIEW

File: lib/api/src/Cardano/Wallet/Api/Http/Server/Error/AssignReedemers.hs [hunk 1]
```diff
@@ -72,129 +60,3 @@ instance IsRecentEra era => IsServerError (ErrAssignRedeemers era) where
...
-fromAlonzoContextError :: AlonzoContextError era -> ServerError
-fromAlonzoContextError = \case
-    TranslationLogicMissingInput inp ->
-        apiError err400 UnresolvedInputs
-            $ T.unwords
-                [ "The transaction I was given contains inputs I don't know"
-                , "about. Please ensure all foreign inputs are specified as "
-                , "part of the API request. The unknown input is:\n\n"
-                , T.pack $ show inp
-                ]
```
Comment (Anviking) [inline lib/api/src/Cardano/Wallet/Api/Http/Server/Error/AssignReedemers.hs:88]: I think this is basically unreachable anyway.

File: lib/api/src/Cardano/Wallet/Api/Http/Server/Error/AssignReedemers.hs [hunk 2]
```diff
@@ -72,129 +60,3 @@ instance IsRecentEra era => IsServerError (ErrAssignRedeemers era) where
...
-fromBabbageContextError = \case
-    AlonzoContextError e -> fromAlonzoContextError e
-    ByronTxOutInContext txOut ->
-        apiError err400 TranslationByronTxOutInContext
-            $ T.unwords
-                [ "The transaction I was given contains a Byron-style TxOut"
-                , "which is not supported when executing Plutus scripts. "
-                , "The offending TxOut is:\n\n"
-                , T.pack $ show txOut
-                ]
```
Comment (Anviking) [inline lib/api/src/Cardano/Wallet/Api/Http/Server/Error/AssignReedemers.hs:114]: This is not unreachable, but i'd question the worth of maintaining this error mapping now. I don't think anyone uses it.

File: lib/balance-tx/test/spec/Internal/Cardano/Write/Tx/BalanceSpec.hs [hunk 3]
```diff
@@ -24,6 +24,7 @@
 {- HLINT ignore "Use camelCase" -}
 
 {-# OPTIONS_GHC -fno-warn-orphans #-}
+{-# OPTIONS_GHC -fno-warn-deprecations #-}
```
Comment (Anviking) [inline lib/balance-tx/test/spec/Internal/Cardano/Write/Tx/BalanceSpec.hs:30]: Can be addressed in later PR IMO.

File: cabal.project [hunk 4]
```diff
@@ -116,9 +116,9 @@ source-repository-package
 
 source-repository-package
     type: git
-    location: https://github.com/IntersectMBO/cardano-addresses
-    tag: 126964483d188c2362393d2a6eea8c65dfc43097
-    --sha256: 1w152imj28ap5dfdc8x9ly0cy7dn500v7k63vw11f70gvgg6x7my
+    location: https://github.com/Anviking/cardano-addresses
```
Comment (Anviking) [inline cabal.project]: Oops. Will fix

File: lib/address-derivation-discovery/lib/Cardano/Wallet/Address/Derivation/SharedKey.hs [hunk 5]
```diff
@@ -147,6 +147,7 @@ constructAddressFromIx role pTemplate dTemplate ix =
         Nothing ->
             createEnterpriseAddress pScript
 
+-- | NOTE: The roles 'DRep', 'CCCold', 'CCHot' are not supported.
```
Comment (Anviking) [inline lib/address-derivation-discovery/lib/Cardano/Wallet/Address/Derivation/SharedKey.hs:150]: Although, I think restricting the domain with a new type containing a subset of `CA.Role` should be fairly painless. May try...

**Review by paolino — APPROVED** [review]
LGTM

**Pull request merged**